File: DPBO-praktikum/4/cpp/Mahasiswa.cpp
// memanggil file header
#include "header.hh"

// constructor kosong
Mahasiswa::Mahasiswa(){
    
}

// construktor berparameter
Mahasiswa::Mahasiswa(string nik,string nim, string nama, string jenis_kelamin, string prodi, string fakultas, string asal_universitas, string email_edu)
: SivitasAkademik(nik, nama, jenis_kelamin, asal_universitas, email_edu) {
this->nim = nim;
this->fakultas = fakultas;
this->prodi = prodi;
}

// ambil nilai nim
string Mahasiswa::get_nim() {
    return nim;
}

// mengisi nilai nim
void Mahasiswa::set_nim(string nim) {
    this->nim = nim;
}

// ambil nilai fakultas
string Mahasiswa::get_fakultas() {
    return fakultas;
}

// mengisi nilai fakultas
void Mahasiswa::set_fakultas(string fakultas) {
    this->fakultas = fakultas;
}

// ambil nilai prodi
string Mahasiswa::get_prodi() {
    return prodi;
}

// mengisi nilai prodi
void Mahasiswa::set_prodi(string prodi) {
    this->prodi = prodi;
}
// destructor
Mahasiswa::~Mahasiswa(){

}

File: DPBO-praktikum/4/python/Mahasiswa.py
from SivitasAkademik import SivitasAkademik

class Mahasiswa(SivitasAkademik):
    def __init__(self, nim, nama, jenis_kelamin, fakultas, asal_universitas, email_edu):
        super().__init__(None, nama, jenis_kelamin, asal_universitas, email_edu)
        self._nim = nim
        self._fakultas = fakultas
        self._prodi = None

    def get_nim(self):
        return self._nim

    def set_nim(self, nim):
        self._nim = nim

    def get_fakultas(self):
        return self._fakultas

    def set_fakultas(self, fakultas):
        self._fakultas = fakultas

    def get_prodi(self):
        return self._prodi

    def set_prodi(self, prodi):
        self._prodi = prodi